app/controllers/internal_api/v1/pillars_controller.rb

```diff
@@ -0,0 +1,45 @@
+# Serve the pillar information
+class InternalApi::V1::PillarsController < InternalApiController
+  def show
+    pillar = Hash[Pillar.pluck :pillar, :value]
+    # rubocop:disable Style/AlignHash
+    ok content: {
+      dashboard: pillar["dashboard"],
+      api: {
+        server: {
+          external_fqdn: pillar["api:server:external_fqdn"]
+        },
+        cluster_ip: pillar["api:cluster_ip"]
+      },
+      cluster_cidr:     pillar["cluster_cidr"],
+      cluster_cidr_min: pillar["cluster_cidr_min"],
+      cluster_cidr_max: pillar["cluster_cidr_max"],
+      cluster_cidr_len: pillar["cluster_cidr_len"],
+      services_cidr:    pillar["services_cidr"],
+      dns: {
+        cluster_ip: pillar["dns:cluster_ip"]
+      },
+      proxy: {
+        systemwide: pillar["proxy:systemwide"],
+        http:       pillar["proxy:http"],
+        https:      pillar["proxy:https"],
+        no_proxy:   pillar["proxy:no_proxy"]
+      },
+      ldap: {
+        host:             pillar["ldap:host"],
+        port:             pillar["ldap:port"],
+        bind_dn:          pillar["ldap:bind_dn"],
+        bind_pw:          pillar["ldap:bind_pw"],
+        domain:           pillar["ldap:domain"],
+        group_dn:         pillar["ldap:group_dn"],
+        people_dn:        pillar["ldap:people_dn"],
+        base_dn:          pillar["ldap:base_dn"],
+        admin_group_dn:   pillar["ldap:admin_group_dn"],
+        admin_group_name: pillar["ldap:admin_group_name"],
+        tls_method:       pillar["ldap:tls_method"],
+        mail_attribute:   pillar["ldap:mail_attribute"]
+      }
+    }
+    # rubocop:enable Style/AlignHash
```

Comment:
I would personally prefer to move that to a dedicated view instead of embedding a whole json object into the controller.

However this is just a matter of personal taste.

Reply:
if it grows more I think this is a good idea. we could even use something like jbuilder, but for now i think it's still fine.